r/R/dataset-write.R

```diff
@@ -54,6 +54,40 @@
 #' - `null_fallback`: character to be used in place of missing values (`NA` or
 #' `NULL`) when using Hive-style partitioning. See [hive_partition()].
 #' @return The input `dataset`, invisibly
+#' @examples
+#' # We can partition by one more variables, say, cyl and gear in mtcars dataset
+#' one_part_dir <- tempfile()
+#' two_part_dir <- tempfile()
+#' write_dataset(mtcars, one_part_dir, "parquet", partitioning = "cyl")
+#' write_dataset(mtcars, two_part_dir, "parquet", partitioning = c("cyl", "gear"))
+#'
+#' # See the contents of the directory we wrote to in order to demonstrate
+#' # what partitioning does
+#' list(
+#'  cyl_partioning = list.files(one_part_dir, pattern = "parquet",
+#'   recursive = TRUE),
+#'  cyl_gear_partitioning = list.files(two_part_dir, pattern = "parquet",
+#'   recursive = TRUE)
+#' )
+#'
+#' # We can do the same combining both arrow and dplyr
+#' if(requireNamespace("dplyr", quietly = TRUE)) {
+#'  two_part_dir_2 <- tempfile()
+#'  two_part_dir_3 <- tempfile()
+#'  d <- mtcars %>% group_by(cyl, gear)
+#'  d %>% write_dataset(two_part_dir_2, "parquet")
+#'
+#'  # Passing the additional hive_style option to see the difference in the
+#'  # output
```

Comment:
```suggestion
#'  # We can also turn off the Hive-style directory naming where the column name is included with the value for each directory with `hive_style = FALSE`.
```

Reply:
This might be outside of the scope of this example (since it concentrates on writing) but it might be nice to add in an example of how one would need to call `read_dataset()` when one has specified `hive_style=FALSE` and how that differs from reading a dataset that does have Hive-style directories.